Anime-App/Feature/Anime/ViewModel/AnimeViewModel.swift

```diff
@@ -8,39 +8,74 @@
 import Foundation
 
 class AnimeViewModel {
-        
+    
     private(set) var animes: [Anime] = []
+    private var queryObject = QueryObject()
+
  
+    func fetchAnime(page: Int, completion: @escaping (Result<Void,NetworkingError>) -> Void) {
+        
+        guard var urlComps = URLComponents(string:  BaseUrl.baseUrl.url + UrlEndpoint.getAnime.endpoint + UrlEndpoint.pagination.endpoint) else { return }
+
+
+        urlComps.queryItems = queryObject.paginationQuery(pageValue: page)
+        guard let animeUrl = urlComps.url else { return }
+        let animeRequest = URLRequest(url: animeUrl)
+        guard let animeAbsoluteString = animeRequest.url?.absoluteString else { return }
+
+        NetworkingManager.shared.request(animeAbsoluteString, type: AnimeResult.self) { res in
```

Comment:
You may want to refactor your networking manager to take in a URL and not a URL string